src/k8s/pkg/k8sd/controllers/feature.go

```diff
@@ -239,27 +240,38 @@ func (c *FeatureController) reconcileLoop(
 
 // isBlocked checks if the feature controller is blocked by an in-progress upgrade.
 // If an upgrade is in progress, the feature controller will not apply any configuration changes.
-func (c *FeatureController) isBlocked(ctx context.Context) (bool, error) {
+func (c *FeatureController) isBlocked(ctx context.Context, getClusterConfig func(context.Context) (types.ClusterConfig, error)) (bool, error) {
 	log := log.FromContext(ctx)
-	k8sClient, err := c.snap.KubernetesClient("")
-	if err != nil {
-		return false, fmt.Errorf("failed to get Kubernetes client: %w", err)
-	}
 
-	upgrade, err := k8sClient.GetInProgressUpgrade(ctx)
+	clusterConfig, err := getClusterConfig(ctx)
 	if err != nil {
-		return false, fmt.Errorf("failed to check for in-progress upgrade: %w", err)
+		return false, fmt.Errorf("failed to retrieve cluster configuration: %w", err)
 	}
+	// Skip feature reconciliation during upgrades to prevent conflicts if cluster configuration
+	// is modified while the upgrade is in progress.
```

Comment:
This comment is a bit confusing to me. I read it as:
```suggestion
	// Skip feature reconciliation while an upgrade is in progress to avoid conflicting cluster
	// configuration changes.
```
... in this case, the snap is unblocking the controller (skipping feature reconciliation) by returning false, but the post-upgrade hook won't execute the reconciliation either due to the user annotation override, right?

Reply:
Yes, the controller is blocked if:
1. an upgrade is in progress
2. the annotation is not set